CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

IntegralProgrammer opened the pull request:
This _Pull Request_ adds the `github_publish_os_kernel_version.py` script whose purpose is to periodically execute on a CARDS hosting VM and update a GitHub repository with the version of the Linux kernel that said VM is running.

This _Pull Request_ also updates the security scanning utilities so that if the version of the currently running Linux kernel for a deployment under scan is known via GitHub, any Linux kernel vulnerabilities detected in non-running kernels are safely ignored.

Testing can be done by following the _updated_ instructions for this (`CARDS-2262`) branch under `Utilities/SecurityScanning/README.md`

IntegralProgrammer committed:
CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

Add the writeGitHubTextFile method to the GitHubRepoHandler module for storing a string in a file on GitHub

IntegralProgrammer committed:
CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

Add the github_publish_os_kernel_version.py utility

IntegralProgrammer committed:
CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

Add the github_get_os_kernel_version.py utility

IntegralProgrammer committed:
CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

Improve the trivy_to_slack.py utility to optionally ignore kernel vulnerabilities in kernels that are not currently running

IntegralProgrammer committed:
CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

Improve scan_remote_package_list.sh to ignore vulerabilities in non-running kernels if the currently running kernel version can be determined

IntegralProgrammer committed:
CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

Fix repeated --running_kernel_version bug in scan_remote_package_list.sh

IntegralProgrammer committed:
CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

Update Utilities/SecurityScanning/README.md

sdumitriu commented on Utilities/SecurityScanning/GitHubRepoHandler.py:
```diff
@@ -115,6 +115,24 @@ def readGitHubTextFile(self, path):
 			raise Exception("HTTP {} was returned when attempting to download {}.".format(resp.status_code, path))
 		return resp.text
 
+	def writeGitHubTextFile(self, path, text_content):
```
What happens if the previous version doesn't exist?

IntegralProgrammer replied:
If the previous version doesn't exist, it's because the file itself doesn't exist and in that case, most likely a `KeyError` would be thrown as the response from GitHub wouldn't contain a `sha` object in its JSON response.

I have renamed this method to `updateGitHubTextFile` in https://github.com/data-team-uhn/cards/pull/1533/commits/aad029575cf917acefa468f2c132047ad46ea808 to reflect the fact that the method can only be used for updating an already existing text file as opposed to creating a new one.

IntegralProgrammer committed:
CARDS-2262: Publish the currently running kernel version for the VM so that vulnerabilities in non-running kernels can be safely ignored

Rename writeGitHubTextFile to updateGitHubTextFile to reflect the fact that the method is only suitable for updating an already existing text file on GitHub as opposed to creating a new one
